packages/core/src/stateUtils.ts

```diff
@@ -609,8 +548,9 @@ export function getInitialState<
  */
 export function getStateNode<TContext, TEvent extends EventObject>(
   stateNode: StateNode<TContext, any, TEvent>,
-  stateKey: string
-): StateNode<TContext, any, TEvent> {
+  stateKey: string,
+  discardInvalidNodes: boolean = false
```

Comment:
I don't like this flag at all, will look into removing it in the future. This has been introduced solely to pass this test:
https://github.com/davidkpiano/xstate/blob/797a53ad88767f649580628380bbdf86e9fc8a0c/packages/core/test/invalid.test.ts#L50
I'm not convinced if we should bother with this within core algorithms - it would be way easier if we could just assume valid values/configurations. If one migrates their schema they should be responsible for resolving conflicts - especially because we can't resolve all of them and a decision on how to migrate removed regions would be better done on the developer's side.

Reply:
@davidkpiano thoughts on this one? should we bother "fixing" invalid inputs in the core?

Reply:
We shouldn't, you're right. Let's assume valid values/configs.